GTG/core/timer.py

```diff
@@ -70,15 +72,59 @@ def time_changed(self):
         self.timeout_source = GObject.timeout_add_seconds(secs_to_refresh,
                                                           self.emit_refresh)
 
-    def set_configuration(self, refresh_hour, refresh_min):
+    def set_configuration(self, time):
+        self.config.set('hour', time.hour)
+        self.config.set('min', time.minute)
+
+    def get_configuration(self):
         try:
-            datetime.time(int(refresh_hour), int(refresh_min))
-        except (ValueError, TypeError):
-            raise ValueError("{}:{} is not a valid time".format(refresh_hour,
-                                                                refresh_min))
+            return datetime.time(int(self.config.get('hour')),
+                                 int(self.config.get('min')))
+        except(ValueError):
+            Log.error("Invalid time values. Try using HH:MM or am/pm format")
+            return datetime.time(0, 0)
 
-        self.config.set('hour', refresh_hour)
-        self.config.set('min', refresh_min)
+    def parse_time(self, time):
+        """
+        This function parses user input when setting
+        start of the day in general preferences.
+        input:  @time               user's input as string
 
-    def get_configuration(self):
-        return self.config.get('hour'), self.config.get('min')
+        output: @hour_value         default format for hours
+                @minute_value       default format for minutes
+                @ValueError         raised if @time is not valid
+        """
+        hour_value = ''
+        minute_value = ''
+        correct_hour, correct_min = False, False
+        # Format for HH:MM and optional am/pm
+        match = re.match(r'^(?P<hours>[0-9]{1,2}):?(?P<minutes>[0-9]{2})?( )?(?P<am_pm>[ap]m)?$', time)
```

Comment:
This line is too ling and causes Travis Cl fail.

@izidormatusov , I tried to apply your advice and used:

 match = re.match(r'^(?P<hours>[0-9]{1,2}):?(?P<minutes>[0-9]{2})?$'\
r'^( )?(?P<am_pm>[ap]m)?$', time)

to split the line and make the travis pass. However, now my test fails some of the cases from test_timer.py (no logically separable group fails, values like 0000 or 5:20 fail :/ )

Am I doing something wrong? Or how about saving the RE to a separate variable and then use only something like:

match = re.match(variable, time)?? What do you think?